### DISCUSSION TITLE
Default values for definitions

### DISCUSSION THREAD

**Discussion by asdine** [cue-lang/cue#1904] [Q&A] [answered]
Is there a way to define a default value for entire structs (and not only certain fields)?

I have been trying like this ([play link](https://cuelang.org/play/?id=drAOetV-e-P#cue@export@cue)):

```cue
#A: {
    foo?: string
    bar?: string
}

a: #A | *{
    foo: "hello"
}

a: {
    bar: "hi"
}

// expected
// a: {bar: "hi"}
// actual:
// a: {foo: "hello", bar: "hi"}
```

**Comment by verdverm** [comment, marked as answer]
You need to fully specify the default, or at least make it not ambiguous

In this case, add `bar?: _|_` to the default. In the original, the default aligns with the definition and the final value.

https://cuelang.org/play/?id=0qwS4y6WXm3#cue@export@cue

**Reply by asdine** [reply]
If we remove the last bit, the default no longer works unfortunately: https://cuelang.org/play/?id=_qMtHGvLfyY#cue@export@cue

**Reply by verdverm** [reply]
```cue
#A: {
    foo?: string
    bar?: string
} | *{
    foo: "hello"
    bar?: _|_
}

a: #A 
```

or

```cue
#A: {
    foo?: string
    bar?: string
} | *{
    foo: "hello"
}
```

depending on the downstream behavior you want

**Reply by asdine** [reply]
Thanks!

**Comment by myitcv** [comment]
> Is there a way to define a default value for entire structs (and not only certain fields)?

I interpreted this question slightly differently:

https://cuelang.org/play/?id=MNSw68kOvEa#cue@export@cue

I'm not clear the use of a disjunction is what's intended here. 

Please let me know if I misunderstood though.
